src/sources/splunk_hec.rs

```diff
@@ -611,6 +629,14 @@ fn raw_event(
         log.insert(log_schema().host_key(), remote.to_string());
     }
 
+    // Add X-Forwarded-For header as remote_addr, only when it can be parsed to IpAddr
+    if let Some(remote_address) = xff {
+        let parsed_address: Result<IpAddr, AddrParseError> = remote_address.parse();
+        if let Ok(_) = parsed_address {
+            log.insert(REMOTE_ADDR, remote_address);
+        }
+    }
```

Comment:
I'd honestly be fine just inserting it as-is rather than checking it is an IP first.

Reply:
If you'd prefer! I just figured it might be a slightly more [rfc](https://datatracker.ietf.org/doc/html/rfc7239#section-6) compliant approach. Though re-reading this section, we're also dropping the port segments too, so that might not be a good leg for me to stand on anyways 😆 

The primary thing I was trying to avoid with this was if someone was trying to jam things into the header that _weren't_ valid xff values, to just drop them. Though I could see how _that_ would be unexpected too